core/src/main/java/org/apache/spark/SparkFirehoseListener.java

```diff
@@ -140,6 +140,16 @@ public void onBlockUpdated(SparkListenerBlockUpdated blockUpdated) {
   }
 
   @Override
+  public void onSpeculativeTaskSubmitted(SparkListenerSpeculativeTaskSubmitted speculativeTask) {
+    onEvent(speculativeTask);
+  }
+
+  @Override
+  public void onExtraExecutorNeeded() {
```

Comment:
Maybe I'm missing something obvious, but what's use of this except for adding some log?